File: module-02-navigation/js/navigation_screenshot.js
// 範例：Playwright 多分頁 + 批量截圖
const { chromium } = require('playwright');

(async () => {
  const browser = await chromium.launch({ headless: true });
  const context = await browser.newContext();
  const urls = [
    'https://example.com/page1',
    'https://example.com/page2',
    'https://example.com/page3'
  ];

  for (let i = 0; i < urls.length; i++) {
    const page = await context.newPage();
    await page.goto(urls[i]);
    const info = await page.locator('div > p:nth-of-type(2) > a');
    await info.screenshot({ path: `screenshots/playwright_info_${i+1}.png` });
    await page.close();
  }

  await browser.close();
})();

File: module-02-navigation/py/navigation_screenshot.py
# 範例：Selenium 多分頁 + 截圖
from selenium import webdriver
from selenium.webdriver.chrome.service import Service
from selenium.webdriver.common.by import By
from webdriver_manager.chrome import ChromeDriverManager

service = Service(ChromeDriverManager().install())
options = webdriver.ChromeOptions()
#options.add_argument('--headless')
driver = webdriver.Chrome(service=service, options=options)

urls = [
    'https://example.com/page1',
    'https://example.com/page2',
    'https://example.com/page3'
]

for idx, url in enumerate(urls, start=1):
    driver.execute_script('window.open()')           # 開新分頁
    driver.switch_to.window(driver.window_handles[-1])
    driver.get(url)
    element = driver.find_element(By.CSS_SELECTOR, 'div > p:nth-of-type(2) > a')
    element.screenshot(f"screenshots/selenium_info_{idx}.png")

# 回到第一分頁
driver.switch_to.window(driver.window_handles[0])
driver.quit()
